src/java.desktop/unix/native/libawt_xawt/xawt/XlibWrapper.c

```diff
@@ -2299,6 +2299,8 @@ Java_sun_awt_X11_XlibWrapper_SetBitmapShape
 
     pRect = (RECT_T *)SAFE_SIZE_ARRAY_ALLOC(malloc, worstBufferSize, sizeof(RECT_T));
     if (!pRect) {
+        (*env)->ReleaseIntArrayElements(env, bitmap, values, JNI_ABORT);
+        JNU_ThrowOutOfMemoryError(env, "malloc failed");
```

Comment:
I think the throw is unnecessary. You will get some incorrect rendering, but that's all.

Reply:
So should I remove the  JNU_ThrowOutOfMemoryError ?  The calloc in the same file has a JNU_ThrowOutOfMemoryError (but maybe the situation is different there?)  .

Reply:
Fair question. That other case is used only during initialisation and if they fail then some things are not set up properly.
So different enough. Yes, I think you can remove the one you are adding. If it ever happens it could be recoverable.